src/services/user-service.js

```diff
@@ -124,6 +137,30 @@ class UserService {
     return user.lockedOut;
   }
 
+  async updateUserStoragePlan(userId, storagePlanId) {
+    logger.info(`Updating storage plan for user with id ${userId}: ${storagePlanId}`);
+
+    const user = await this.getUserById(userId);
+    if (!user) {
+      throw new NotFound(`User with ID '${userId}' could not be found`);
+    }
+
+    const plan = await this.getStoragePlanById(storagePlanId);
+    if (!plan) {
+      throw new NotFound(`Storage plan with ID '${storagePlanId}' could not be found`);
+    }
+
+    const oldStorage = user.storage || { plan: null, usedStorageInBytes: 0, reminders: [] };
+    if (oldStorage.plan) {
```

Comment:
I had this first, but it requires an eslint exception because of "redundant" braces (`{ ...() }`). I agree on the `getDefaultStorage`, I will change it.

Reply:
Done.